diesel/src/mysql/types/mod.rs

```diff
@@ -4,7 +4,34 @@ mod date_and_time;
 use mysql::{Mysql, MysqlType};
 use std::error::Error as StdError;
 use std::io::Write;
-use types::{ToSql, IsNull, FromSql, HasSqlType};
+use types::{ToSql, IsNull, NotNull, FromSql, HasSqlType};
+
+
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq, PartialOrd, Ord)] pub struct Tinyint;
+
+impl NotNull for Tinyint {}
+
+impl ToSql<Tinyint, Mysql> for i8 {
+    fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<StdError+Send+Sync>> {
+        let int_value:i32 = *self as i32;
+        <i32 as ToSql<::types::Integer, Mysql>>::to_sql(&int_value, out)
+    }
+}
+
+impl FromSql<Tinyint, Mysql> for i8 {
+    fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<StdError+Send+Sync>> {
+        let bytes = not_none!(bytes);
+        Ok(bytes[0] as i8)
+    }
+}
+
+impl HasSqlType<Tinyint> for Mysql {
+    fn metadata() -> MysqlType {
+        MysqlType::Tiny
+    }
+}
+
+//primitive_impls!(Tinyint -> (i8, mysql: (Tiny)));
```

Comment:
To make this work, you need to bring `TinyInt` in `diesel::types`. You can have a look at how we did that for pg here: https://github.com/diesel-rs/diesel/blob/master/diesel/src/types/mod.rs#L273

For that you need to make the types module from mysql public too